exec/cpg.c

```diff
@@ -663,59 +663,68 @@ static int notify_lib_joinlist(
 	int size;
 	char *buf;
 	struct qb_list_head *iter;
-	int count;
+	int member_list_entries;
 	struct res_lib_cpg_confchg_callback *res;
 	mar_cpg_address_t *retgi;
+	int i;
 
-	count = 0;
+	/*
+	 * Find size of member_list (use process_info_list but remove items in left_list)
+	 */
+	member_list_entries = 0;
 
 	qb_list_for_each(iter, &process_info_list_head) {
 		struct process_info *pi = qb_list_entry (iter, struct process_info, list);
+
 		if (mar_name_compare (&pi->group, group_name) == 0) {
-			int i;
-			int founded = 0;
+			int in_left_list = 0;
 
 			for (i = 0; i < left_list_entries; i++) {
 				if (left_list[i].nodeid == pi->nodeid && left_list[i].pid == pi->pid) {
-					founded++;
+					in_left_list = 1;
+					break ;
 				}
 			}
 
-			if (!founded)
-				count++;
+			if (!in_left_list) {
+				member_list_entries++;
+			}
 		}
 	}
 
```

Comment:
this and the loop to fill member_list below are identical except for what happens in the last ```if``` - maybe something like

```
int some_fitting_name(const mar_cpg_name_t *group_name, int left_list_entries, mar_cpg_address_t *left_list, mar_cpg_address_t *retgi) {

int member_list_entries = 0;
qb_list_for_each(iter, &process_info_list_head) {
	struct process_info *pi = qb_list_entry (iter, struct process_info, list);

	if (mar_name_compare (&pi->group, group_name) == 0) {
		int in_left_list = 0;

		for (i = 0; i < left_list_entries; i++) {
			if (left_list[i].nodeid == pi->nodeid && left_list[i].pid == pi->pid) {
				in_left_list = 1;
				break ;
			}
		}

		if (!in_left_list) {
			member_list_entries++;

			if (retgi !+ NULL) {
				retgi->nodeid = pi->nodeid;
				retgi->pid = pi->pid;
				retgi->reason = CPG_REASON_UNDEFINED;
				retgi++;
			}
		}
	}

	return member_list_entries;
}
```

would allow this to be replaced by
``` member_list_entries = some_fitting_name(group_name, left_list_entries, left_list, NULL);```

and the one below by ```retgi += some_fitting_name(group_name, left_list_entries, left_list, retgi);```?

maybe I am just too used to re-using code blocks like that from perl though ;)

Reply:
Actually I was thinking about it (before your comment), but I had really problem with finding `some_fitting_name`. Anyway, I gave it a try so it is in next patch (but I'm not too impressed by git stat)